rune_trie.go

```diff
@@ -19,11 +19,17 @@ func NewRuneTrie() *RuneTrie {
 // nodes or for nodes with a value of nil.
 func (trie *RuneTrie) Get(key string) interface{} {
 	node := trie
-	for _, r := range key {
+	prevNode := node
+	for idx, r := range key {
 		node = node.children[r]
 		if node == nil {
+			// sfx additional condition to return prev value
+			if idx > 0 && key[idx-1] == '.' {
```

Comment:
`i have used "*" as the value; so that if we have to do extra matching (in future) during Get() we can use this special char. Also because we are returning prevNode.value, (if needed) we can use that to check that prefix is indeed of type mention in above example.`

so, why are you checking for last being a "." here when we're never ever doing that. we're putting "*" not "."